app/models/container_deployment.rb

```diff
@@ -0,0 +1,11 @@
+class ContainerDeployment < ApplicationRecord
+  has_many :container_deployment_authentications
+  belongs_to :deployed_ems, :class_name => 'ExtManagementSystem'
+  belongs_to :deployed_on_ems, :class_name => 'ExtManagementSystem'
```

Comment:
you might set `inverse_of:` here.
Also on `ExtManagementSystem` I only see one `has_many :container_deployments`